---
title: "BLAST Workflow Template" 
author: "Author: Daniela Cassol (danielac@ucr.edu) and Thomas Girke (thomas.girke@ucr.edu)"
date: "Last update: `r format(Sys.time(), '%d %B, %Y')`" 
output:
  BiocStyle::html_document:
    toc_float: true
    code_folding: show
package: systemPipeR
vignette: |
  %\VignetteIndexEntry{SPblast Workflow Template}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
fontsize: 14pt
bibliography: bibtex.bib
---

<!--
- Compile from command-line
Rscript -e "rmarkdown::render('SPblast.Rmd', c('BiocStyle::html_document'), clean=TRUE)"
-->

```{css, echo=FALSE}
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
```

```{r style, echo = FALSE, results = 'asis'}
BiocStyle::markdown()
options(width=60, max.print=1000)
knitr::opts_chunk$set(
    eval=as.logical(Sys.getenv("KNITR_EVAL", "TRUE")),
    cache=as.logical(Sys.getenv("KNITR_CACHE", "TRUE")), 
    tidy.opts=list(width.cutoff=60), tidy=TRUE)
```

```{r setup, echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
suppressPackageStartupMessages({
    library(systemPipeR)
    library(SPblast)
})
```

# Introduction

The [Basic Local Alignment Search Tool (BLAST)](https://blast.ncbi.nlm.nih.gov/Blast.cgi) finds regions of local similarity between sequences. 

| BLAST Tool | Query Type | Database Type | Description                                                 |
|------------|------------|---------------|-------------------------------------------------------------|
| `blastn`     | Nucleotide | Nucleotide    | Search a nucleotide database using a nucleotide query       |
| `blastx`     | Nucleotide | Protein       | Search protein database using a translated nucleotide query |
| `tblastn`    | Protein    | Nucleotide    | Search translated nucleotide database using a protein query |
| `blastp`     | Protein    | Protein       | Search protein database using a protein query               |

## Experimental design

Typically, users want to specify here all information relevant for the analysis
of their NGS study. This includes detailed descriptions of FASTQ files,
experimental design, reference genome, gene annotations, etc. 

# Workflow environment

## Load packages and sample data

The `systemPipeR` package needs to be loaded to perform the analysis 
steps shown in this report [@H_Backman2016-bt]. The package allows users
to run the entire analysis workflow interactively or with a single command 
while also generating the corresponding analysis report. For details
see `systemPipeR's` main [vignette](http://www.bioconductor.org/packages/devel/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html).

```{r load_systempiper, eval=TRUE, message=FALSE, warning=FALSE}
library(systemPipeR)
library(SPblast)
```

## Setup and Requirements

To go through this tutorial, you need the following software installed:

* R/>=4.0.3
* systemPipeR R package (version 1.24)
* blastp: 2.9.0+

If you desire to build your pipeline with any different software, make sure to have the respective software installed and configured in your PATH. To make sure if the configuration is right, you always can test as follow:

```{r tryCL, eval=TRUE}
tryCL(command = "blastp")  
```

## Generate workflow environment

[*systemPipeRdata*](http://bioconductor.org/packages/release/data/experiment/html/systemPipeRdata.html) package is a helper package to generate a fully populated [*systemPipeR*](http://bioconductor.org/packages/release/bioc/html/systemPipeR.html)
workflow environment in the current working directory with a single command. 
All the instruction for generating the workflow template are provide in the *systemPipeRdata* vignette [here](http://www.bioconductor.org/packages/devel/data/experiment/vignettes/systemPipeRdata/inst/doc/systemPipeRdata.html#1_Introduction). 

After building and loading the workflow environment generated by `genWorkenvir` 
from *systemPipeRdata* all data inputs are stored in
a `data/` directory and all analysis results will be written to a separate
`results/` directory, while the `SPblast.Rmd` script and the `targets` file are expected to be located in the parent directory. The R session is expected to run from this parent
directory. Additional parameter files are stored under `param/`.

To work with real data, users want to organize their own data similarly
and substitute all test data for their own data. To rerun an established
workflow on new data, the initial `targets` file along with the corresponding
FASTQ files are usually the only inputs the user needs to provide.

For more details, please consult the documentation 
[here](http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html). More information about the `targets` files from *systemPipeR* can be found [here](http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#25_structure_of_targets_file). 

## Get Data

We are using for this tutorial mouse and human [RefSeq](ftp://ftp.ncbi.nih.gov/refseq/) 
protein data sets from NCBI. 
First, we download the required dataset to `data/` folder and decompressing files, as follows:

```{r download.data, eval=FALSE}
download.file("ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.1.protein.faa.gz", 
              destfile = "data/mouse.1.protein.faa.gz")
R.utils::gunzip("data/mouse.1.protein.faa.gz")
# download.file("ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.2.protein.faa.gz", 
#               destfile = "data/mouse.2.protein.faa.gz")
# R.utils::gunzip("data/mouse.2.protein.faa.gz")
download.file("ftp://ftp.ncbi.nih.gov/refseq/H_sapiens/mRNA_Prot/human.1.protein.faa.gz",
              destfile = "data/human.1.protein.faa.gz")
R.utils::gunzip("data/human.1.protein.faa.gz")
```


```{r subset_data, eval=FALSE}
library(Biostrings)
mouse.path <- system.file("extdata", "data/mouse.1.protein.faa", package="SPblast")
mouse.path <- "data/mouse.1.protein.faa"
select <- readAAStringSet(mouse.path)
select <- select[1:3]
writeXStringSet(select, "data/mouse_select.fasta", format="fasta")
```

## Targets

## Create workflow from command line

```{r, eval=FALSE}
# makeblastdb -in ./data/<dataset>.faa -out ./data/<dataset>.faa -dbtype prot -hash_index -parse_seqids

baseCommand <- list(baseCommand="makeblastdb")
inputs <- list(
  "input_file"=list(type="File", preF="-in", yml="_INPUT_FASTA_"),
  "database_name"= list(type="string", preF="-out", yml="_DATABASE_NAME_"),
  "molecule_type"= list(type="string", preF="-dbtype", yml="_DBTYPE_")  )
arguments <- list(
 "hash_index"=list(preF="-hash_index"),
  "parse_seqids"=list(preF="-parse_seqids"))
outputs <- list("out.phd"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).phd"),
                "out.phi"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).phd"),
                "out.phr"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).phr"),
                "out.pin"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).pin"),
                "out.pog"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).pog"),
                "out.psd"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).psd"),
                "out.psi"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).psi"),
                "out.psq"=list(type="File", "$(inputs.results_path)/$(inputs.database_name.basename).psq")
                )
commandLine <- list(baseCommand=baseCommand, arguments=arguments, inputs=inputs, outputs=outputs)


## Create a SYSargs2 object and populate all the command-line
WF <- createWF(targets=NULL, commandLine, results_path="./data", module_load="ncbi-blast",
               file = "default", overwrite = TRUE, cwlVersion = "v1.0")

param <- list(input_file=list(class="File", path="./data/human.1.protein.faa"), 
              database_name="./data/human.1.protein.faa", 
              molecule_type="prot")
WF <- config.param(input_file=WF, param, file="default")

WF <- renderWF(WF, inputvars=c(input_file="./data/mouse.1.protein.faa", database_name="./data/mouse.1.protein.faa",
                               molecule_type="prot"))
cmdlist(WF)
output(WF)

# blastp -query ./data/<myseq>.fasta -db ./data/<dataset>.faa -outfmt 0 -evalue 1e-6 -out <blastp.out>

baseCommand <- list(baseCommand="blastp")
inputs <- list(
  "query"=list(type="File", preF="-query", yml="_INPUT_FASTA_"),
  "database_name"= list(type="File", preF="-db", yml="_DATABASE_NAME_"),
  "evalue"=list(type="int", preF="-evalue", yml="_evalue_"),
  "outfmt"= list(type="int", preF="-outfmt", yml="_outfmt_"),
  "out"=list(type="File", preF="-out", yml="_OUTPUT_"))
outputs <- list("out"=list(type="File", "$(inputs.results_path)/$(inputs.out)"))
commandLine <- list(baseCommand=baseCommand, inputs=inputs, outputs=outputs)

## Create a SYSargs2 object and populate all the command-line
WF <- createWF(targets=NULL, commandLine, results_path="./results", module_load="ncbi-blast",
               file = "default", overwrite = TRUE, cwlVersion = "v1.0")
param <- list(query=list(class="File", path="./data/mouse_select.fasta"), 
              database_name=list(class="File", path="./data/human.1.protein.faa"), 
              evalue=1e-6,
              outfmt=0L,
              out=list(class="File", path="blastp.out"))
WF <- config.param(input_file=WF, param, file="default")

WF <- renderWF(WF)
cmdlist(WF)
output(WF)

# blastp -query ./data/<myseq>.fasta -db ./data/<dataset>.faa -outfmt 6 -evalue 1e-6 -out <blastp.tab>
WF <- loadWF(targets=NULL, wf_file="blastp.cwl", input_file="blastp.yml", dir_path="param/cwl/blastp/")
param <- list(query=list(class="File", path="./data/mouse_select.fasta"), 
              database_name=list(class="File", path="./data/human.1.protein.faa"), 
              evalue="1e-6",
              outfmt=6L,
              out=list(class="File", path="results/blastp.tab"))
WF <- config.param(input_file=WF, param, file="default")

WF <- renderWF(WF)
cmdlist(WF)
output(WF)
```

## Run workflow

Now open the R markdown script `SPblast.Rmd`in your R IDE (_e.g._ vim-r or RStudio) and 
run the workflow as outlined below. 

Here pair-end workflow example is provided. Please refer to the main vignette 
`systemPipeR.Rmd` for running the workflow with single-end data. 

If you are running on a single machine, use following code as an example to check 
if some tools used in this workflow are in your environment **PATH**. No warning 
message should be shown if all tools are installed.

## Experiment definition provided by `targets` file

The `targets` file defines all FASTQ files and sample comparisons of the analysis workflow. 

```{r load_targets, eval=TRUE}
targetspath <- system.file("extdata", "targetsPE.txt", package="systemPipeR")
targets <- read.delim(targetspath, comment.char = "#")[,1:4]
targets
```

```{r, eval=FALSE}
dir_path <- system.file("extdata", "param/cwl/makeblastdb/", package="SPblast")
WF <- loadWF(targets=NULL, wf_file="makeblastdb.cwl", input_file="makeblastdb.yml",
             dir_path=dir_path)
param <- list(input_file=list(class="File", path="./data/human.1.protein.faa"), 
              database_name="./data/human.1.protein.faa", 
              molecule_type="prot")
WF <- config.param(input_file=WF, param, file="default")

WF <- renderWF(WF, inputvars=c(input_file="./data/mouse.1.protein.faa", database_name="./data/mouse.1.protein.faa",
                               molecule_type="prot"))
cmdlist(WF)
output(WF)
runCommandline(WF, make_bam = FALSE)
```


```{r, eval=FALSE}
dir_path <- system.file("extdata", "param/cwl/blastp/", package="SPblast")
WF <- loadWF(targets=NULL, wf_file="blastp.cwl", input_file="blastp.yml", dir_path=dir_path)
param <- list(query=list(class="File", path="./data/mouse_select.fasta"), 
              database_name=list(class="File", path="./data/human.1.protein.faa"), 
              evalue="1e-6",
              outfmt=6L,
              out=list(class="File", path="results/blastp.tab"))
WF <- config.param(input_file=WF, param, file="default")

WF <- renderWF(WF)
cmdlist(WF)
output(WF)
runCommandline(WF, make_bam = FALSE)
```

# Version Information

```{r sessionInfo}
sessionInfo()
```

# Funding

This project is funded by NSF award ABI-1661152.

# References
