<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BLAST Workflow Template • systemPipeBLAST</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="BLAST Workflow Template">
<meta property="og:description" content="systemPipeBLAST">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-xxxx"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-xxxx');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeBLAST</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/systemPipeBLAST.html">
    <span class="fas fa-book"></span>
     
    Workflow
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/systemPipeBLAST">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="systemPipeBLAST_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BLAST Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 08 February, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/systemPipeBLAST/blob/master/vignettes/systemPipeBLAST.Rmd"><code>vignettes/systemPipeBLAST.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeBLAST.Rmd</code></div>

    </div>

    
    
<!--
- Compile from command-line
Rscript -e "rmarkdown::render('systemPipeBLAST.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeBLAST.Rmd', tangle=TRUE)"; Rscript -e "rmarkdown::render('systemPipeBLAST.Rmd', c('BiocStyle::pdf_document'))"
-->
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-design" class="anchor"></a>Experimental design</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div id="workflow-environment" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-environment" class="anchor"></a>Workflow environment</h1>
<div id="load-packages-and-sample-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages-and-sample-data" class="anchor"></a>Load packages and sample data</h2>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>. The package allows users to run the entire analysis workflow interactively or with a single command while also generating the corresponding analysis report. For details see <code>systemPipeR's</code> main <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html">vignette</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://girke.bioinformatics.ucr.edu/systemPipeR/">systemPipeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://systempipe.org">systemPipeBLAST</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="setup-and-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#setup-and-requirements" class="anchor"></a>Setup and Requirements</h2>
<p>To go through this tutorial, you need the following software installed:</p>
<ul>
<li>R/&gt;=4.0.3</li>
<li>systemPipeR R package (version 1.24)</li>
<li>blastp: 2.9.0+</li>
</ul>
<p>If you desire to build your pipeline with any different software, make sure to have the respective software installed and configured in your PATH. To make sure if the configuration is right, you always can test as follow:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/tryCL.html">tryCL</a></span><span class="op">(</span>command <span class="op">=</span> <span class="st">"blastp"</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "All set up, proceed!"</code></pre>
</div>
<div id="generate-workflow-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-workflow-environment" class="anchor"></a>Generate workflow environment</h2>
<p><a href="http://bioconductor.org/packages/release/data/experiment/html/systemPipeRdata.html"><em>systemPipeRdata</em></a> package is a helper package to generate a fully populated <a href="http://bioconductor.org/packages/release/bioc/html/systemPipeR.html"><em>systemPipeR</em></a> workflow environment in the current working directory with a single command. All the instruction for generating the workflow template are provide in the <em>systemPipeRdata</em> vignette <a href="http://www.bioconductor.org/packages/devel/data/experiment/vignettes/systemPipeRdata/inst/doc/systemPipeRdata.html#1_Introduction">here</a>.</p>
<p>After building and loading the workflow environment generated by <code>genWorkenvir</code> from <em>systemPipeRdata</em> all data inputs are stored in a <code>data/</code> directory and all analysis results will be written to a separate <code>results/</code> directory, while the <code>systemPipeBLAST.Rmd</code> script and the <code>targets</code> file are expected to be located in the parent directory. The R session is expected to run from this parent directory. Additional parameter files are stored under <code>param/</code>.</p>
<p>To work with real data, users want to organize their own data similarly and substitute all test data for their own data. To rerun an established workflow on new data, the initial <code>targets</code> file along with the corresponding FASTQ files are usually the only inputs the user needs to provide.</p>
<p>For more details, please consult the documentation <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html">here</a>. More information about the <code>targets</code> files from <em>systemPipeR</em> can be found <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#25_structure_of_targets_file">here</a>.</p>
</div>
<div id="get-data" class="section level2">
<h2 class="hasAnchor">
<a href="#get-data" class="anchor"></a>Get Data</h2>
<p>We are using for this tutorial mouse and human <a href="ftp://ftp.ncbi.nih.gov/refseq/">RefSeq</a> protein data sets from NCBI. First, we download the required dataset to <code>data/</code> folder and decompressing files, as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.1.protein.faa.gz"</span>, 
    destfile <span class="op">=</span> <span class="st">"data/mouse.1.protein.faa.gz"</span><span class="op">)</span>
<span class="fu">R.utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/R.utils/man/compressFile.html">gunzip</a></span><span class="op">(</span><span class="st">"data/mouse.1.protein.faa.gz"</span><span class="op">)</span>
<span class="co"># download.file('ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.2.protein.faa.gz',</span>
<span class="co"># destfile = 'data/mouse.2.protein.faa.gz')</span>
<span class="co"># R.utils::gunzip('data/mouse.2.protein.faa.gz')</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ncbi.nih.gov/refseq/H_sapiens/mRNA_Prot/human.1.protein.faa.gz"</span>, 
    destfile <span class="op">=</span> <span class="st">"data/human.1.protein.faa.gz"</span><span class="op">)</span>
<span class="fu">R.utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/R.utils/man/compressFile.html">gunzip</a></span><span class="op">(</span><span class="st">"data/human.1.protein.faa.gz"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="va">mouse.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"data/mouse.1.protein.faa"</span>, 
    package <span class="op">=</span> <span class="st">"systemPipeBLAST"</span><span class="op">)</span>
<span class="va">select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readAAStringSet</a></span><span class="op">(</span><span class="st">"mouse.path"</span><span class="op">)</span>
<span class="va">select</span> <span class="op">&lt;-</span> <span class="va">select</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">writeXStringSet</a></span><span class="op">(</span><span class="va">select</span>, <span class="st">"data/mouse_select.fasta"</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></code></pre></div>
</div>
<div id="targets" class="section level2">
<h2 class="hasAnchor">
<a href="#targets" class="anchor"></a>Targets</h2>
</div>
<div id="create-workflow-from-command-line" class="section level2">
<h2 class="hasAnchor">
<a href="#create-workflow-from-command-line" class="anchor"></a>Create workflow from command line</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># makeblastdb -in ./data/&lt;dataset&gt;.faa -out</span>
<span class="co"># ./data/&lt;dataset&gt;.faa -dbtype prot -hash_index -parse_seqids</span>

<span class="va">baseCommand</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>baseCommand <span class="op">=</span> <span class="st">"makeblastdb"</span><span class="op">)</span>
<span class="va">inputs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, preF <span class="op">=</span> <span class="st">"-in"</span>, 
    yml <span class="op">=</span> <span class="st">"_INPUT_FASTA_"</span><span class="op">)</span>, database_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"string"</span>, 
    preF <span class="op">=</span> <span class="st">"-out"</span>, yml <span class="op">=</span> <span class="st">"_DATABASE_NAME_"</span><span class="op">)</span>, molecule_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"string"</span>, 
    preF <span class="op">=</span> <span class="st">"-dbtype"</span>, yml <span class="op">=</span> <span class="st">"_DBTYPE_"</span><span class="op">)</span><span class="op">)</span>
<span class="va">arguments</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>hash_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>preF <span class="op">=</span> <span class="st">"-hash_index"</span><span class="op">)</span>, parse_seqids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>preF <span class="op">=</span> <span class="st">"-parse_seqids"</span><span class="op">)</span><span class="op">)</span>
<span class="va">outputs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>out.phd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).phd"</span><span class="op">)</span>, 
    out.phi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).phd"</span><span class="op">)</span>, 
    out.phr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).phr"</span><span class="op">)</span>, 
    out.pin <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).pin"</span><span class="op">)</span>, 
    out.pog <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).pog"</span><span class="op">)</span>, 
    out.psd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).psd"</span><span class="op">)</span>, 
    out.psi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).psi"</span><span class="op">)</span>, 
    out.psq <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.database_name.basename).psq"</span><span class="op">)</span><span class="op">)</span>
<span class="va">commandLine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>baseCommand <span class="op">=</span> <span class="va">baseCommand</span>, arguments <span class="op">=</span> <span class="va">arguments</span>, 
    inputs <span class="op">=</span> <span class="va">inputs</span>, outputs <span class="op">=</span> <span class="va">outputs</span><span class="op">)</span>


<span class="co">## Create a SYSargs2 object and populate all the command-line</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/createWF.html">createWF</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">commandLine</span>, results_path <span class="op">=</span> <span class="st">"./data"</span>, 
    module_load <span class="op">=</span> <span class="st">"ncbi-blast"</span>, file <span class="op">=</span> <span class="st">"default"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>, 
    cwlVersion <span class="op">=</span> <span class="st">"v1.0"</span><span class="op">)</span>

<span class="va">param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span>, molecule_type <span class="op">=</span> <span class="st">"prot"</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/config.param.html">config.param</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">WF</span>, <span class="va">param</span>, file <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span><span class="op">(</span><span class="va">WF</span>, inputvars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="st">"./data/mouse.1.protein.faa"</span>, 
    database_name <span class="op">=</span> <span class="st">"./data/mouse.1.protein.faa"</span>, molecule_type <span class="op">=</span> <span class="st">"prot"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">cmdlist</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>

<span class="co"># blastp -query ./data/&lt;myseq&gt;.fasta -db ./data/&lt;dataset&gt;.faa</span>
<span class="co"># -outfmt 0 -evalue 1e-6 -out &lt;blastp.out&gt;</span>

<span class="va">baseCommand</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>baseCommand <span class="op">=</span> <span class="st">"blastp"</span><span class="op">)</span>
<span class="va">inputs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, preF <span class="op">=</span> <span class="st">"-query"</span>, yml <span class="op">=</span> <span class="st">"_INPUT_FASTA_"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, preF <span class="op">=</span> <span class="st">"-db"</span>, yml <span class="op">=</span> <span class="st">"_DATABASE_NAME_"</span><span class="op">)</span>, 
    evalue <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"int"</span>, preF <span class="op">=</span> <span class="st">"-evalue"</span>, yml <span class="op">=</span> <span class="st">"_evalue_"</span><span class="op">)</span>, 
    outfmt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"int"</span>, preF <span class="op">=</span> <span class="st">"-outfmt"</span>, yml <span class="op">=</span> <span class="st">"_outfmt_"</span><span class="op">)</span>, 
    out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, preF <span class="op">=</span> <span class="st">"-out"</span>, yml <span class="op">=</span> <span class="st">"_OUTPUT_"</span><span class="op">)</span><span class="op">)</span>
<span class="va">outputs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"File"</span>, <span class="st">"$(inputs.results_path)/$(inputs.out)"</span><span class="op">)</span><span class="op">)</span>
<span class="va">commandLine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>baseCommand <span class="op">=</span> <span class="va">baseCommand</span>, inputs <span class="op">=</span> <span class="va">inputs</span>, 
    outputs <span class="op">=</span> <span class="va">outputs</span><span class="op">)</span>

<span class="co">## Create a SYSargs2 object and populate all the command-line</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/createWF.html">createWF</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">commandLine</span>, results_path <span class="op">=</span> <span class="st">"./results"</span>, 
    module_load <span class="op">=</span> <span class="st">"ncbi-blast"</span>, file <span class="op">=</span> <span class="st">"default"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>, 
    cwlVersion <span class="op">=</span> <span class="st">"v1.0"</span><span class="op">)</span>
<span class="va">param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/mouse_select.fasta"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span><span class="op">)</span>, 
    evalue <span class="op">=</span> <span class="fl">1e-06</span>, outfmt <span class="op">=</span> <span class="fl">0L</span>, out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"blastp.out"</span><span class="op">)</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/config.param.html">config.param</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">WF</span>, <span class="va">param</span>, file <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">cmdlist</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>

<span class="co"># blastp -query ./data/&lt;myseq&gt;.fasta -db ./data/&lt;dataset&gt;.faa</span>
<span class="co"># -outfmt 6 -evalue 1e-6 -out &lt;blastp.tab&gt;</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="cn">NULL</span>, wf_file <span class="op">=</span> <span class="st">"blastp.cwl"</span>, input_file <span class="op">=</span> <span class="st">"blastp.yml"</span>, 
    dir_path <span class="op">=</span> <span class="st">"param/cwl/blastp/"</span><span class="op">)</span>
<span class="va">param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/mouse_select.fasta"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span><span class="op">)</span>, 
    evalue <span class="op">=</span> <span class="st">"1e-6"</span>, outfmt <span class="op">=</span> <span class="fl">6L</span>, out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, 
        path <span class="op">=</span> <span class="st">"results/blastp.tab"</span><span class="op">)</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/config.param.html">config.param</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">WF</span>, <span class="va">param</span>, file <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">cmdlist</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span></code></pre></div>
</div>
<div id="run-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#run-workflow" class="anchor"></a>Run workflow</h2>
<p>Now open the R markdown script <code>systemPipeBLAST.Rmd</code>in your R IDE (<em>e.g.</em> vim-r or RStudio) and run the workflow as outlined below.</p>
<p>Here pair-end workflow example is provided. Please refer to the main vignette <code>systemPipeR.Rmd</code> for running the workflow with single-end data.</p>
<p>If you are running on a single machine, use following code as an example to check if some tools used in this workflow are in your environment <strong>PATH</strong>. No warning message should be shown if all tools are installed.</p>
</div>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">targetspath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"targetsPE.txt"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">targetspath</span>, comment.char <span class="op">=</span> <span class="st">"#"</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="va">targets</span></code></pre></div>
<pre><code>##                      FileName1                   FileName2
## 1  ./data/SRR446027_1.fastq.gz ./data/SRR446027_2.fastq.gz
## 2  ./data/SRR446028_1.fastq.gz ./data/SRR446028_2.fastq.gz
## 3  ./data/SRR446029_1.fastq.gz ./data/SRR446029_2.fastq.gz
## 4  ./data/SRR446030_1.fastq.gz ./data/SRR446030_2.fastq.gz
## 5  ./data/SRR446031_1.fastq.gz ./data/SRR446031_2.fastq.gz
## 6  ./data/SRR446032_1.fastq.gz ./data/SRR446032_2.fastq.gz
## 7  ./data/SRR446033_1.fastq.gz ./data/SRR446033_2.fastq.gz
## 8  ./data/SRR446034_1.fastq.gz ./data/SRR446034_2.fastq.gz
## 9  ./data/SRR446035_1.fastq.gz ./data/SRR446035_2.fastq.gz
## 10 ./data/SRR446036_1.fastq.gz ./data/SRR446036_2.fastq.gz
## 11 ./data/SRR446037_1.fastq.gz ./data/SRR446037_2.fastq.gz
## 12 ./data/SRR446038_1.fastq.gz ./data/SRR446038_2.fastq.gz
## 13 ./data/SRR446039_1.fastq.gz ./data/SRR446039_2.fastq.gz
## 14 ./data/SRR446040_1.fastq.gz ./data/SRR446040_2.fastq.gz
## 15 ./data/SRR446041_1.fastq.gz ./data/SRR446041_2.fastq.gz
## 16 ./data/SRR446042_1.fastq.gz ./data/SRR446042_2.fastq.gz
## 17 ./data/SRR446043_1.fastq.gz ./data/SRR446043_2.fastq.gz
## 18 ./data/SRR446044_1.fastq.gz ./data/SRR446044_2.fastq.gz
##    SampleName Factor
## 1         M1A     M1
## 2         M1B     M1
## 3         A1A     A1
## 4         A1B     A1
## 5         V1A     V1
## 6         V1B     V1
## 7         M6A     M6
## 8         M6B     M6
## 9         A6A     A6
## 10        A6B     A6
## 11        V6A     V6
## 12        V6B     V6
## 13       M12A    M12
## 14       M12B    M12
## 15       A12A    A12
## 16       A12B    A12
## 17       V12A    V12
## 18       V12B    V12</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dir_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"param/cwl/makeblastdb/"</span>, 
    package <span class="op">=</span> <span class="st">"systemPipeBLAST"</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="cn">NULL</span>, wf_file <span class="op">=</span> <span class="st">"makeblastdb.cwl"</span>, input_file <span class="op">=</span> <span class="st">"makeblastdb.yml"</span>, 
    dir_path <span class="op">=</span> <span class="va">dir_path</span><span class="op">)</span>
<span class="va">param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span>, molecule_type <span class="op">=</span> <span class="st">"prot"</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/config.param.html">config.param</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">WF</span>, <span class="va">param</span>, file <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span><span class="op">(</span><span class="va">WF</span>, inputvars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="st">"./data/mouse.1.protein.faa"</span>, 
    database_name <span class="op">=</span> <span class="st">"./data/mouse.1.protein.faa"</span>, molecule_type <span class="op">=</span> <span class="st">"prot"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">cmdlist</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">WF</span>, make_bam <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dir_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"param/cwl/blastp/"</span>, package <span class="op">=</span> <span class="st">"systemPipeBLAST"</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="cn">NULL</span>, wf_file <span class="op">=</span> <span class="st">"blastp.cwl"</span>, input_file <span class="op">=</span> <span class="st">"blastp.yml"</span>, 
    dir_path <span class="op">=</span> <span class="va">dir_path</span><span class="op">)</span>
<span class="va">param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/mouse_select.fasta"</span><span class="op">)</span>, 
    database_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, path <span class="op">=</span> <span class="st">"./data/human.1.protein.faa"</span><span class="op">)</span>, 
    evalue <span class="op">=</span> <span class="st">"1e-6"</span>, outfmt <span class="op">=</span> <span class="fl">6L</span>, out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"File"</span>, 
        path <span class="op">=</span> <span class="st">"results/blastp.tab"</span><span class="op">)</span><span class="op">)</span>
<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/config.param.html">config.param</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">WF</span>, <span class="va">param</span>, file <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span>

<span class="va">WF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">cmdlist</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs2-methods.html">output</a></span><span class="op">(</span><span class="va">WF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span><span class="op">(</span><span class="va">WF</span>, make_bam <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /home/dcassol/src/R-4.0.3/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices
## [6] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] systemPipeBLAST_0.9.0       systemPipeR_1.25.5         
##  [3] ShortRead_1.48.0            GenomicAlignments_1.26.0   
##  [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
##  [7] MatrixGenerics_1.2.1        matrixStats_0.58.0         
##  [9] BiocParallel_1.24.1         Rsamtools_2.6.0            
## [11] Biostrings_2.58.0           XVector_0.30.0             
## [13] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
## [15] IRanges_2.24.1              S4Vectors_0.28.1           
## [17] BiocGenerics_0.36.0         BiocStyle_2.18.1           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0         rjson_0.2.20            
##   [3] hwriter_1.3.2            ellipsis_0.3.1          
##   [5] rprojroot_2.0.2          fs_1.5.0                
##   [7] remotes_2.2.0            bit64_4.0.5             
##   [9] AnnotationDbi_1.52.0     xml2_1.3.2              
##  [11] codetools_0.2-18         splines_4.0.3           
##  [13] cachem_1.0.3             knitr_1.31              
##  [15] jsonlite_1.7.2           systemPipeRdata_1.18.1  
##  [17] annotate_1.68.0          GO.db_3.12.1            
##  [19] dbplyr_2.1.0             png_0.1-7               
##  [21] pheatmap_1.0.12          graph_1.68.0            
##  [23] BiocManager_1.30.10      compiler_4.0.3          
##  [25] httr_1.4.2               backports_1.2.1         
##  [27] GOstats_2.56.0           assertthat_0.2.1        
##  [29] Matrix_1.3-2             fastmap_1.1.0           
##  [31] limma_3.46.0             formatR_1.7             
##  [33] htmltools_0.5.1.1        prettyunits_1.1.1       
##  [35] tools_4.0.3              gtable_0.3.0            
##  [37] glue_1.4.2               GenomeInfoDbData_1.2.4  
##  [39] Category_2.56.0          dplyr_1.0.4             
##  [41] rsvg_2.1                 batchtools_0.9.15       
##  [43] rappdirs_0.3.3           V8_3.4.0                
##  [45] Rcpp_1.0.6               pkgdown_1.6.1           
##  [47] vctrs_0.3.6              rtracklayer_1.50.0      
##  [49] xfun_0.20                stringr_1.4.0           
##  [51] lifecycle_0.2.0          XML_3.99-0.5            
##  [53] edgeR_3.32.1             zlibbioc_1.36.0         
##  [55] scales_1.1.1             BSgenome_1.58.0         
##  [57] VariantAnnotation_1.36.0 ragg_0.4.1              
##  [59] hms_1.0.0                RBGL_1.66.0             
##  [61] RColorBrewer_1.1-2       yaml_2.2.1              
##  [63] curl_4.3                 memoise_2.0.0           
##  [65] ggplot2_3.3.3            biomaRt_2.46.2          
##  [67] latticeExtra_0.6-29      stringi_1.5.3           
##  [69] RSQLite_2.2.3            genefilter_1.72.1       
##  [71] desc_1.2.0               checkmate_2.0.0         
##  [73] GenomicFeatures_1.42.1   DOT_0.1                 
##  [75] rlang_0.4.10             pkgconfig_2.0.3         
##  [77] systemfonts_1.0.0        bitops_1.0-6            
##  [79] evaluate_0.14            lattice_0.20-41         
##  [81] purrr_0.3.4              bit_4.0.4               
##  [83] tidyselect_1.1.0         GSEABase_1.52.1         
##  [85] AnnotationForge_1.32.0   magrittr_2.0.1          
##  [87] bookdown_0.21            R6_2.5.0                
##  [89] generics_0.1.0           base64url_1.4           
##  [91] DelayedArray_0.16.1      DBI_1.1.1               
##  [93] withr_2.4.1              pillar_1.4.7            
##  [95] survival_3.2-7           RCurl_1.98-1.2          
##  [97] tibble_3.0.6             crayon_1.4.0            
##  [99] BiocFileCache_1.14.0     rmarkdown_2.6           
## [101] jpeg_0.1-8.1             progress_1.2.2          
## [103] locfit_1.5-9.4           grid_4.0.3              
## [105] data.table_1.13.6        blob_1.2.1              
## [107] Rgraphviz_2.34.0         digest_0.6.27           
## [109] xtable_1.8-4             brew_1.0-6              
## [111] textshaping_0.2.1        openssl_1.4.3           
## [113] munsell_0.5.0            askpass_1.1</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This research was funded by National Science Foundation Grants IOS-0750811 and MCB-1021969, and a Marie Curie European Economic Community Fellowship PIOF-GA-2012-327954.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-H_Backman2016-bt" class="csl-entry">
H Backman, Tyler W, and Thomas Girke. 2016. <span>“<span class="nocase">systemPipeR: NGS workflow and report generation environment</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '',
    indexName: '',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
